server/sheet.go

```diff
@@ -67,6 +73,211 @@ func (s *Server) registerSheetRoutes(g *echo.Group) {
+				file.Path,
+				repo.BranchFilter,
+			)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch file meta from VCS, instance URL: %s, repo ID: %s, file path: %s, branch: %s", vcs.InstanceURL, repo.ExternalID, file.Path, repo.BranchFilter)).SetInternal(err)
+			}
+
+			fileContent, err := vcsPlugin.Get(vcs.Type, vcsPlugin.ProviderConfig{Logger: s.l}).ReadFileContent(ctx,
+				common.OauthContext{
+					ClientID:     vcs.ApplicationID,
+					ClientSecret: vcs.Secret,
+					AccessToken:  repo.AccessToken,
+					RefreshToken: repo.RefreshToken,
+					Refresher:    s.refreshToken(ctx, repo.ID),
+				},
+				vcs.InstanceURL,
+				repo.ExternalID,
+				file.Path,
+				repo.BranchFilter,
+			)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch file content from VCS, instance URL: %s, repo ID: %s, file path: %s, branch: %s", vcs.InstanceURL, repo.ExternalID, file.Path, repo.BranchFilter)).SetInternal(err)
+			}
+
+			lastCommit, err := vcsPlugin.Get(vcs.Type, vcsPlugin.ProviderConfig{Logger: s.l}).FetchCommitByID(ctx,
+				common.OauthContext{
+					ClientID:     vcs.ApplicationID,
+					ClientSecret: vcs.Secret,
+					AccessToken:  repo.AccessToken,
+					RefreshToken: repo.RefreshToken,
+					Refresher:    s.refreshToken(ctx, repo.ID),
+				},
+				vcs.InstanceURL,
+				repo.ExternalID,
+				fileMeta.LastCommitID,
+			)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch commit data from VCS, instance URL: %s, repo ID: %s, commit ID: %s", vcs.InstanceURL, repo.ExternalID, fileMeta.LastCommitID)).SetInternal(err)
+			}
+
+			sheetVCSPayload := &api.SheetVCSPayload{
+				FileName:     fileMeta.Name,
+				FilePath:     fileMeta.Path,
+				Size:         fileMeta.Size,
+				Author:       lastCommit.AuthorName,
+				LastCommitID: lastCommit.ID,
+				LastSyncTs:   time.Now().Unix(),
+			}
+			payload, err := json.Marshal(sheetVCSPayload)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to marshal sheetVCSPayload").SetInternal(err)
+			}
+
+			payloadString := string(payload)
+
+			var sheetSource api.SheetSource
+			switch vcs.Type {
+			case vcsPlugin.GitLabSelfHost:
+				sheetSource = api.SheetFromGitLabSelfHost
+			case vcsPlugin.GitHubCom:
+				sheetSource = api.SheetFromGitHubCom
+			}
+			vscSheetType := api.SheetForSQL
+			sheet, err := s.SheetService.FindSheet(ctx, &api.SheetFind{
+				Name:      &sheetInfo.SheetName,
+				ProjectID: &project.ID,
+				Source:    &sheetSource,
+				Type:      &vscSheetType,
+			})
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to find sheet with name: %s, project ID: %d", sheetInfo.SheetName, projectID)).SetInternal(err)
+			}
+
+			var databaseID *int
+			// In non-tenant mode, we can set a databaseId for sheet with ENV_NAME and DB_NAME.
```

Comment:
Looks like we require ENV_NAME and DB_NAME must be present at the same time.

What if only DB_NAME is present, or do just simply not allow it?

Reply:
We cannot get a unique database by DB_NAME, so we need a ENV_NAME to get a special instance.
`CREATE UNIQUE INDEX idx_db_unique_instance_id_name ON db(instance_id, name);`

Reply:
That's reasonable, so we can leave a comment saying ENV_NAME and DB_NAME is either both present or neither present.